### ISSUE TITLE
workflows.argoproj.io/completed is false when output set for resource creation

### ISSUE THREAD

**Issue by mcullenEST** [argoproj/argo-workflows#13085]
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I have tested with the `:latest` image tag (i.e. `quay.io/argoproj/workflow-controller:latest`) and can confirm the issue still exists on `:latest`. If not, I have explained why, **in detail**, in my description below.
- [X] I have searched existing issues and could not find a match for this bug
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/main/docs/CONTRIBUTING.md))

### What happened/what did you expect to happen?

When running a workflow that uses a resource creates and returns a value. the workflows.argoproj.io/completed remains as false even though workflows.argoproj.io/phase=Succeeded.

In my case, I dynamically create a pvc and I want to return the pvc name so other templates can mount it (as shown in the workflow before). The logic works and the correct result is achieved. I have also tried with other k8 manifests and I get the same results. Running without the output sets workflows.argoproj.io/completed to true. So the issue is with the output section.

Even returning a simple string like "abc" from the resource template still causes workflows.argoproj.io/completed remains as false.

Tested on Argo Workflow 3.5.4

### Version

3.5.4

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: output-parameter-
spec:
  entrypoint: main
  templates:
    - name: main
      steps:        
        - - name: create-pvc
            template: create-pvc
     
    - name: create-pvc
      resource:
        action: create
        setOwnerReference: true
        manifest: |
          apiVersion: v1
          kind: PersistentVolumeClaim
          metadata:
            generateName: "my-pvc-"
          spec:
            accessModes:
              - "ReadWriteMany"
            resources:
              requests:
                storage: "1Gi"
      outputs:
        parameters:
          - name: pvc-name
            # value: "abc" #Also doesn't work
            valueFrom:
             jqFilter: '.metadata.name'
```

### Logs from the workflow controller

```text
time="2024-05-24T13:46:13.507Z" level=info msg="Processing workflow" Phase= ResourceVersion=66771886 namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:13.516Z" level=info msg="Task-result reconciliation" namespace=sandbox numObjs=0 workflow=output-parameter-drf2m
time="2024-05-24T13:46:13.516Z" level=info msg="Updated phase  -> Running" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:13.516Z" level=warning msg="Node was nil, will be initialized as type Skipped" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:13.517Z" level=info msg="was unable to obtain node for , letting display name to be nodeName" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:13.517Z" level=info msg="Steps node output-parameter-drf2m initialized Running" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:13.517Z" level=info msg="StepGroup node output-parameter-drf2m-800642104 initialized Running" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:13.517Z" level=warning msg="Node was nil, will be initialized as type Skipped" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:13.518Z" level=info msg="Pod node output-parameter-drf2m-2509746212 initialized Pending" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:13.536Z" level=info msg="Created pod: output-parameter-drf2m[0].create-pvc (output-parameter-drf2m-create-pvc-2509746212)" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:13.536Z" level=info msg="Workflow step group node output-parameter-drf2m-800642104 not yet completed" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:13.536Z" level=info msg="TaskSet Reconciliation" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:13.536Z" level=info msg=reconcileAgentPod namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:13.547Z" level=info msg="Workflow update successful" namespace=sandbox phase=Running resourceVersion=66771890 workflow=output-parameter-drf2m
time="2024-05-24T13:46:23.537Z" level=info msg="Processing workflow" Phase=Running ResourceVersion=66771890 namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:23.538Z" level=info msg="Task-result reconciliation" namespace=sandbox numObjs=1 workflow=output-parameter-drf2m
time="2024-05-24T13:46:23.538Z" level=info msg="task-result changed" namespace=sandbox nodeID=output-parameter-drf2m-2509746212 workflow=output-parameter-drf2m
time="2024-05-24T13:46:23.538Z" level=info msg="node changed" namespace=sandbox new.message= new.phase=Succeeded new.progress=0/1 nodeID=output-parameter-drf2m-2509746212 old.message= old.phase=Pending old.progress=0/1 workflow=output-parameter-drf2m
time="2024-05-24T13:46:23.539Z" level=info msg="Step group node output-parameter-drf2m-800642104 successful" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:23.539Z" level=info msg="node output-parameter-drf2m-800642104 phase Running -> Succeeded" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:23.539Z" level=info msg="node output-parameter-drf2m-800642104 finished: 2024-05-24 13:46:23.539255187 +0000 UTC" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:23.539Z" level=info msg="Outbound nodes of output-parameter-drf2m-2509746212 is [output-parameter-drf2m-2509746212]" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:23.539Z" level=info msg="Outbound nodes of output-parameter-drf2m is [output-parameter-drf2m-2509746212]" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:23.539Z" level=info msg="node output-parameter-drf2m phase Running -> Succeeded" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:23.539Z" level=info msg="node output-parameter-drf2m finished: 2024-05-24 13:46:23.53942472 +0000 UTC" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:23.539Z" level=info msg="TaskSet Reconciliation" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:23.539Z" level=info msg=reconcileAgentPod namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:23.539Z" level=info msg="Updated phase Running -> Succeeded" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:23.545Z" level=info msg="cleaning up pod" action=deletePod key=sandbox/output-parameter-drf2m-1340600742-agent/deletePod
time="2024-05-24T13:46:23.548Z" level=info msg="Workflow update successful" namespace=sandbox phase=Succeeded resourceVersion=66771952 workflow=output-parameter-drf2m
time="2024-05-24T13:46:23.556Z" level=info msg="cleaning up pod" action=labelPodCompleted key=sandbox/output-parameter-drf2m-create-pvc-2509746212/labelPodCompleted
time="2024-05-24T13:46:33.551Z" level=info msg="Processing workflow" Phase=Succeeded ResourceVersion=66771952 namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:33.551Z" level=info msg="Task-result reconciliation" namespace=sandbox numObjs=1 workflow=output-parameter-drf2m
time="2024-05-24T13:46:33.551Z" level=info msg="task-result changed" namespace=sandbox nodeID=output-parameter-drf2m-2509746212 workflow=output-parameter-drf2m
time="2024-05-24T13:46:33.551Z" level=info msg="TaskSet Reconciliation" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:33.551Z" level=info msg=reconcileAgentPod namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:46:33.557Z" level=info msg="cleaning up pod" action=deletePod key=sandbox/output-parameter-drf2m-1340600742-agent/deletePod
time="2024-05-24T13:46:33.564Z" level=info msg="Workflow update successful" namespace=sandbox phase=Succeeded resourceVersion=66771952 workflow=output-parameter-drf2m
time="2024-05-24T13:50:39.028Z" level=info msg="Processing workflow" Phase=Succeeded ResourceVersion=66771952 namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:50:39.028Z" level=info msg="Task-result reconciliation" namespace=sandbox numObjs=1 workflow=output-parameter-drf2m
time="2024-05-24T13:50:39.028Z" level=info msg="task-result changed" namespace=sandbox nodeID=output-parameter-drf2m-2509746212 workflow=output-parameter-drf2m
time="2024-05-24T13:50:39.028Z" level=info msg="TaskSet Reconciliation" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:50:39.028Z" level=info msg=reconcileAgentPod namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T13:50:39.408Z" level=info msg="cleaning up pod" action=deletePod key=sandbox/output-parameter-drf2m-1340600742-agent/deletePod
time="2024-05-24T13:50:39.563Z" level=info msg="Workflow update successful" namespace=sandbox phase=Succeeded resourceVersion=66771952 workflow=output-parameter-drf2m
time="2024-05-24T14:00:13.030Z" level=info msg="Processing workflow" Phase=Succeeded ResourceVersion=66771952 namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T14:00:13.030Z" level=info msg="Task-result reconciliation" namespace=sandbox numObjs=1 workflow=output-parameter-drf2m
time="2024-05-24T14:00:13.030Z" level=info msg="task-result changed" namespace=sandbox nodeID=output-parameter-drf2m-2509746212 workflow=output-parameter-drf2m
time="2024-05-24T14:00:13.030Z" level=info msg="TaskSet Reconciliation" namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T14:00:13.030Z" level=info msg=reconcileAgentPod namespace=sandbox workflow=output-parameter-drf2m
time="2024-05-24T14:00:13.363Z" level=info msg="cleaning up pod" action=deletePod key=sandbox/output-parameter-drf2m-1340600742-agent/deletePod
time="2024-05-24T14:00:13.515Z" level=info msg="Workflow update successful" namespace=sandbox phase=Succeeded resourceVersion=66771952 workflow=output-parameter-drf2m
```

### Logs from in your workflow's wait container

```text
kubectl logs -n argo -c wait -l workflows.argoproj.io/workflow=${workflow},workflow.argoproj.io/phase!=Succeeded
```

**Comment by Joibel** [comment]
Can you confirm this is still a problem on 3.5.6 please?

**Comment by mcullenEST** [comment]
resolved in latest release 3.5.7. 
Thanks for speedy response!
